packages/teams-js/test/private/externalAppCardActionsForCEA.spec.ts

```diff
@@ -0,0 +1,226 @@
+          ).rejects.toThrowError(
+            new Error(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedFrameContexts,
+              )}. Current context: "${frameContext}".`,
+            ),
+          );
+        });
+      }
+    });
+  });
+
+  describe('processActionOpenUrl', () => {
+    const allowedFrameContexts = [FrameContexts.content];
+    const testUrl = new URL('https://example.com');
+    const testError = {
+      errorCode: ActionOpenUrlErrorCode.INTERNAL_ERROR,
+      message: 'testMessage',
+    };
+    const testResponse = ActionOpenUrlType.DeepLinkDialog;
+
+    it('should not allow calls before initialization', async () => {
+      expect.assertions(1);
+      try {
+        await externalAppCardActionsForCEA.processActionOpenUrl(testAppId, testConversationId, testUrl);
+      } catch (e) {
+        expect(e).toEqual(new Error(errorLibraryNotInitialized));
+      }
+    });
+
+    it('should throw error when externalAppCardActionsForCEA capability is not supported', async () => {
+      expect.assertions(1);
+      await utils.initializeWithContext(FrameContexts.content);
+      utils.setRuntimeConfig({ apiVersion: 2, supports: {} });
+
+      try {
+        await externalAppCardActionsForCEA.processActionOpenUrl(testAppId, testConversationId, testUrl);
+      } catch (e) {
+        expect(e).toEqual(errorNotSupportedOnPlatform);
+      }
+    });
+
+    Object.values(FrameContexts).forEach((frameContext) => {
+      if (allowedFrameContexts.includes(frameContext)) {
+        it(`should resolve when called from context - ${frameContext}`, async () => {
+          expect.assertions(3);
+          await utils.initializeWithContext(frameContext);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: { externalAppCardActionsForCEA: {} } });
+
+          const promise = externalAppCardActionsForCEA.processActionOpenUrl(testAppId, testConversationId, testUrl);
+
+          const message = utils.findMessageByFunc(ApiName.ExternalAppCardActionsForCEA_ProcessActionOpenUrl);
+          if (message && message.args) {
+            expect(message).not.toBeNull();
+            expect(message.args).toEqual([testAppId, testUrl.href, testConversationId]);
+            utils.respondToMessage(message, null, testResponse);
+          }
+
+          await expect(promise).resolves.toEqual(testResponse);
+        });
+
+        it(`should throw error from host when called from context - ${frameContext}`, async () => {
+          expect.assertions(3);
+          await utils.initializeWithContext(frameContext);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: { externalAppCardActionsForCEA: {} } });
+
+          const promise = externalAppCardActionsForCEA.processActionOpenUrl(testAppId, testConversationId, testUrl);
+
+          const message = utils.findMessageByFunc('externalAppCardActionsForCEA.processActionOpenUrl');
```

Comment:
>'externalAppCardActionsForCEA.processActionOpenUrl' [](http://example.com/codeflow?start=50&length=51)

Can this reference `ApiName.ExternalAppCardActionsForCEA_ProcessActionOpenUrl` also?